### ISSUE TITLE
Request: on-device applications

### ISSUE THREAD

**Issue by BluBloos** [even-realities/EvenDemoApp#16]
I'm opening this issue to facilitate organized discussion surrounding a future where we can write applications for the G* glasses that run on-device instead of applications over BLE.

In order for the G glasses to be truly open, I believe this is absolutely needed.

I hope other community members feel the same and I would love to hear from others what their desires are

**Comment by icejix** [comment]
I agree, 100%. I think it's absolutely necessary that we're able to upload our own firmware to the glasses, and have access to the hardware itself. Otherwise, this product is not truly "open," but the same as just about every other product with a developer's API.

**Comment by martin-braun** [comment]
I came here for the same reason. I heard them being open-source just to figure out we, so far, have only a demo app that demonstrates how to communicate with the glasses and thats it.

No, that it is not open source. Only opening firmware/drivers of this thing and allowing us to flash it (with our own risk) would make the glasses truly open-source. 

I love the idea of smart glasses and I would like to adopt some in my life, but these glasses can only be affective when they are worn in any situation, making them very privacy invasive. 

**I will NOT carry something on my nose 16 hours per day, if I cannot audit what it's truly doing!**

**Comment by martin-braun** [comment]
@Pingasmaster Maybe my comment sounded harsher than anticipated. I outlined the clear facts of this not being open-source. They truly have all time they need to make it happen, and then, only then, we shall treat it as such.

However, they put out things like:

<img width="255" alt="'Even Realities officially supports Open-Source'" src="https://preview.redd.it/even-realities-officially-supports-open-source-v0-gbbb7uwrcuxd1.jpeg?auto=webp&s=e03399f8fa20639b5a3f184835a46fe215795b4b">

> We're building an open source community to unlock user customization — giving you the power to shape the future of G1. Together, we can create the most optimal features and an even more stunning display.

You apparently aren't ready and never claimed to be, but still creating hype and expectation without really delivering anything truly open-source so far. Let's hope this will age well, then.

**Comment by orcinus** [comment]
> However, they put out things like:

I don't see a problem, they did not post "Even Realities are now Open Source", did they?
And all the other messaging was pretty clear too.
You're overreacting.

**Comment by martin-braun** [comment]
@orcinus 

Did I claim they post "Even Realities are now Open Source"? Their marketing could just be perceived in a wrong way. I wouldn't say they try to earn some sympathy points from the open source world without contributing to it, but statements like these set some expectations in some heads as you probably can tell by looking at the Issues section of this repository in a whole.

I'm not overreacting, don't worry, I'm calm and relaxed. It was just a way to share where I and many here where coming from. Their statement created a hype where none should be. I don't wish to point fingers and so you should not, too.

**Comment by BluBloos** [comment]
Hey guys! Just thinking about this again. Has there been any updates?

**Comment by sllawcj** [comment]
Echoing this. 

Its been a year since they went "open source" and we still don't even have a true SDK available.  

With the release of new hardware with a separate app, we're a hop skip and a jump from this being abadonware. 

This is exactly why the community demanded open-source in the first place.
